cypress/e2e/editing/editing.spec.ts

```diff
@@ -62,4 +63,83 @@ describe("Editing", () => {
+        let editEventId: string;
+
+        // create a second user, which sends, and edits, the event.
+
+        cy.window({ log: false })
+            .then((win) => {
+                cy.getBot(synapse, { displayName: "Bob", userIdPrefix: "bob_" }).then(async (bob) => {
+                    const room = await bob.createRoom({ name: "TestRoom", visibility: win.matrixcs.Visibility.Public });
+                    testRoomId = room.room_id;
+                    cy.log(`Bot user created room ${room.room_id}`);
+
+                    originalEventId = (await bob.sendMessage(room.room_id, { body: "original", msgtype: "m.text" }))
+                        .event_id;
+                    cy.log(`Bot user sent original event ${originalEventId}`);
+
+                    // send a load of padding events. We make them large, so that they fill the whole screen
+                    // and the client doesn't end up paginating into the event we want.
+                    function mkPadding(n: number): MessageEvent {
+                        return MessageEvent.from(`padding ${n}`, `<h3>Test event ${n}</h3>\n`.repeat(10));
+                    }
+                    let i = 0;
+                    while (i < 20) {
+                        const ev = mkPadding(i++);
+                        await bob.sendMessage(room.room_id, ev.serialize().content);
+                    }
+
+                    // ... then the edit ...
+                    editEventId = (
+                        await bob.sendMessage(room.room_id, {
+                            "m.new_content": { body: "Edited body", msgtype: "m.text" },
+                            "m.relates_to": {
+                                rel_type: "m.replace",
+                                event_id: originalEventId,
+                            },
+                            "body": "* edited",
+                            "msgtype": "m.text",
+                        })
+                    ).event_id;
+                    cy.log(`Bot user sent edit event ${editEventId}`);
+
+                    // ... then a load more padding ...
+                    while (i < 40) {
+                        const ev = mkPadding(i++);
+                        await bob.sendMessage(room.room_id, ev.serialize().content);
+                    }
+                });
+            })
+            .then(() => {
+                // now have the cypress user join the room, jump to the original event, and wait for the event to be
+                // visible
+                cy.joinRoom(testRoomId);
+                cy.viewRoomByName("TestRoom");
+                cy.visit(`#/room/${testRoomId}/${originalEventId}`);
+                cy.get(`[data-event-id="${originalEventId}"]`).then((messageTile) => {
+                    // at this point, the edit event should still be unknown
+                    cy.getClient().then((cli) => {
+                        expect(cli.getRoom(testRoomId).getTimelineForEvent(editEventId)).to.be.null;
```

Comment:
question for the reviewer: I'm not very happy with all this indentation; it feels like a return to the bad old days of javascript callback hell. Generally I feel like I only half-grok how Cypress is intended to be used :(. Suggestions as to how to make this better would be welcome.

Reply:
Instead of using `then` to handle the promise, can't we use here `async/await` to handle the promise flow ?

Reply:
```js
await win = cy.window({ log: false })
...
cy.joinRoom(testRoomId)
...
const messageTile = await cy.get(`[data-event-id="${originalEventId}"]`)
const cli = await cy.getClient()
...
```

Reply:
I think it sounds good but doesn't work. The things returned by `cy.*` aren't real promises, so can't be `await`ed. https://github.com/cypress-io/cypress/issues/1417 has a whole bunch to say about this.

Reply:
You might prefer to use an [alias](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Aliases) and then the `cy.get("@aliasname").then(...` to reduce nesting.

Reply:
I've managed to clean this up a fair bit, partly with the use of `cy.all`.

Reply:
Still tick.